antarest/study/business/link_management.py

```diff
@@ -54,8 +69,79 @@ def get_all_links(
+            element: GenericElement = GenericElement(id=area_id, name=area.name)
+            item_list: List[GenericElement] = []
+            for link in area.links:
+                link_element = GenericElement(id=link, name=file_study.config.areas[link].name)
+                item_list.append(
+                    link_element
+                )
+            if len(item_list) > 0:
+                result.append(GenericItem(element=element, item_list=item_list))
+
+        return result
+
+    def get_clusters_list(
+            self, study: Study
+    ) -> List[GenericItem]:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        result = []
+        for area_id, area in file_study.config.areas.items():
+            element: GenericElement = GenericElement(id=area_id, name=area.name)
+            item_list: List[GenericElement] = []
+            for renewable in area.renewables:
+                cluster_element = GenericElement(id=renewable.id, name=renewable.name)
+                item_list.append(
+                    cluster_element
+                )
+            for thermal in area.thermals:
+                cluster_element = GenericElement(id=thermal.id, name=thermal.name)
+                item_list.append(
+                    cluster_element
+                )
+            if len(item_list) > 0:
+                result.append(GenericItem(element=element, item_list=item_list))
+
+        return result
+
+    def get_clusters_and_links(self, study: Study) -> AllCLustersAndLinks:
+        file_study = self.storage_service.get_storage(study).get_raw(study)
+        cluster_result = []
+        link_result = []
+        for area_id, area in file_study.config.areas.items():
+            element: GenericElement = GenericElement(id=area_id, name=area.name)
+            cluster_item_list: List[GenericElement] = []
+            link_item_list: List[GenericElement] = []
+            # for renewable in area.renewables:
+            #    cluster_element = GenericElement(id=renewable.id, name=renewable.name)
+            #    cluster_item_list.append(
+            #        cluster_element
+            #    )
+            for thermal in area.thermals:
+                cluster_element = GenericElement(id=thermal.id, name=thermal.name)
+                cluster_item_list.append(
+                    cluster_element
+                )
+            for link in area.links:
+                link_element = GenericElement(id=link, name=file_study.config.areas[link].name)
+                link_item_list.append(
+                    link_element
+                )
+            if len(cluster_item_list) > 0:
+                cluster_result.append(GenericItem(element=element, item_list=cluster_item_list))
+            if len(link_item_list) > 0:
+                link_result.append(GenericItem(element=element, item_list=link_item_list))
+
+        return AllCLustersAndLinks(links=link_result, clusters=cluster_result)
```

Comment:
Put all this in binding_constraint_manager as this kind of data is used by the binding_constraint form.
Though I don't understand the purpose of these method since you can already get these data in the synthesis